staking/cli/src/instructions.rs

```diff
@@ -870,6 +878,191 @@ pub fn update_y(rpc_client: &RpcClient, signer: &dyn Signer, y: u64) {
+                token_program: spl_token::ID,
+            };
+
+            let data = integrity_pool::instruction::AdvanceDelegationRecord {};
+
+            let ix = Instruction {
+                program_id: integrity_pool::ID,
+                accounts:   accounts.to_account_metas(None),
+                data:       data.data(),
+            };
+
+
+            let mut transaction =
+                Transaction::new_with_payer(&[ix.clone()], Some(&signer.pubkey()));
+            transaction.sign(
+                &[signer],
+                rpc_client
+                    .get_latest_blockhash_with_commitment(CommitmentConfig::finalized())
+                    .unwrap()
+                    .0,
+            );
+            let res = rpc_client.simulate_transaction(&transaction).unwrap();
+            let reward_amount: u64 = u64::from_le_bytes(
+                BASE64_STANDARD
+                    .decode(res.value.return_data.unwrap().data.0)
+                    .unwrap()[..8]
+                    .try_into()
+                    .unwrap(),
+            );
+
+            if reward_amount < min_reward {
+                return;
+            }
+
+            println!(
+                "Advance delegation record for pubkey: {:?} publisher: {:?} with reward: {:?}",
+                positions.to_string(),
+                publisher,
+                reward_amount,
+            );
+
+            process_transaction(rpc_client, &[ix], &[signer]).unwrap();
+        });
+}
+
+pub fn claim_rewards(
+    rpc_client: &RpcClient,
+    signer: &dyn Signer,
+    min_staked: u64,
+    min_reward: u64,
+) {
+    let mut data: Vec<DynamicPositionArrayAccount> = rpc_client
+        .get_program_accounts_with_config(
+            &staking::ID,
+            RpcProgramAccountsConfig {
+                filters:        Some(vec![RpcFilterType::Memcmp(Memcmp::new(
+                    0,
+                    MemcmpEncodedBytes::Bytes(PositionData::discriminator().to_vec()),
+                ))]),
+                account_config: RpcAccountInfoConfig {
+                    encoding:         Some(UiAccountEncoding::Base64Zstd),
+                    data_slice:       None,
+                    commitment:       None,
+                    min_context_slot: None,
+                },
+                with_context:   None,
+            },
+        )
+        .unwrap()
+        .into_iter()
+        .map(|(pubkey, account)| DynamicPositionArrayAccount {
+            key:      pubkey,
+            lamports: account.lamports,
+            data:     account.data.clone(),
+        })
+        .collect::<Vec<_>>();
+
+    let current_epoch = get_current_epoch(rpc_client);
+
+    let data: Vec<DynamicPositionArray> = data
+        .iter_mut()
+        .filter_map(|positions| {
+            let acc = positions.to_dynamic_position_array();
+            let valid = acc
+                .get_target_exposure(&Target::IntegrityPool, current_epoch)
```

Comment:
Not sure about using get_target_exposure here. get_target_exposure ignores UNLOCKED positions but these can have unclaimed rewards